packages/venia-ui/lib/components/Image/image.js

```diff
@@ -75,18 +113,48 @@ const Image = props => {
     );
 };
 
+const conditionallyRequiredString = (props, propName, componentName) => {
+    // This component needs one of src or resource to be provided.
+    if (!props.src && !props.resource) {
```

Comment:
Yeah, there are really two different "use cases" which I'm working on the documentation for:

1. I have a `src`

Pass it to `<Image>` as `src` and it will be used directly.
`srcset` and `sizes` do not come into play here.

2. I have a Magento "resource" - some string returned from GraphQL like "vt-13-main.jpg".

Pass that to `<Image>` as `resource`.
Optionally pass a `resourceWidth` and `resourceHeight` too, that will affect the fallback `src` used by older browsers that don't support `srcset` / `sizes`.
It will create a `srcset` for you.
You should pass in `sizes` depending on your usage.
The browser will pick the best image in `srcset` based on `sizes`, the viewport, and `window.devicePixelRatio`.

So here I'm trying to say "you need to pass at least one of `src` or `resource`".